src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -1755,16 +1754,19 @@ private IVariableDeclarationGroupOperation CreateBoundMultipleLocalDeclarationsO
             SyntaxNode declarationSyntax = declarationGroupSyntax.IsKind(SyntaxKind.LocalDeclarationStatement) ?
                     ((LocalDeclarationStatementSyntax)declarationGroupSyntax).Declaration :
                     declarationGroupSyntax;
+
+            bool isUsing = declarationGroupSyntax.IsKind(SyntaxKind.LocalDeclarationStatement) && ((LocalDeclarationStatementSyntax)declarationGroupSyntax).UsingKeyword != default;
```

Comment:
>declarationGroupSyntax.IsKind(SyntaxKind.LocalDeclarationStatement) && ((LocalDeclarationStatementSyntax)declarationGroupSyntax).UsingKeyword != default; [](start = 27, length = 153)

We shouldn't be relying on syntax in IOperation factory. All required information should be captured in bound nodes during binding and factory should rely only on information in bound nodes.

Reply:
Agreed, but we were already relying on syntax here to distinguish between using/fixed and localdecl. Should we just create an issue to fix the underlying bound nodes (as well as splitting them out above) and link it here?

Reply:
> Should we just create an issue to fix the underlying bound nodes (as well as splitting them out above) and link it here?

I don't think so. Picking best syntax to associate node with is one thing, reinterring semantic meaning (which should be clear from the bound node itself) from syntax is another.

---
In reply to: [333679279](https://github.com/dotnet/roslyn/pull/39125#discussion_r333679279) [](ancestors = 333679279)

Reply:
Fixed.